app/models/repo.rb

```diff
@@ -92,9 +92,9 @@ def populate_issues!
     background_populate_issues(self.id)
   end
 
-  def self.exists_with_name?(name)
+  def self.exists_with_name?(name, repos)
     user_name, repo_name = name.downcase.split(?/)
-    Repo.exists?(user_name: user_name, name: repo_name)
+    repos.map { |r| r.full_name }.include?("#{user_name}/#{repo_name}")
```

Comment:
The original query was also a single query ->

``` ruby
>> Repo.exists?(user_name: 'c', name: 'd').to_sql
  Repo Exists (0.6ms)  SELECT  1 AS one FROM "repos" WHERE "repos"."user_name" = $1 AND "repos"."name" = $2 LIMIT 1  [["user_name", "c"], ["name", "d"]]
```

Here, we are doing calculation in Ruby but I think after adding indexes, the query performance will be better. So I guess original code will perform better with indexes.